lib/Frontend/ModuleInterfaceSupport.cpp

```diff
@@ -124,13 +138,21 @@ static void printImports(raw_ostream &out,
       continue;
     }
 
+    llvm::SmallVector<Identifier, 4> spis;
+    M->lookupImportedSPIGroups(importedModule, spis);
+
+    // Print only implementation-only imports if they have an SPI import.
```

Comment:
Grammar nit: as written, this says that only implementation-only imports get printed (to the exclusion of other imports). I think you mean something more like:

> Only print implementation-only imports which have an SPI import.